app/scripts/controllers/detect-tokens.js

```diff
@@ -182,12 +188,26 @@ export default class DetectTokensController {
    * Restart token detection polling period and call detectNewTokens
    * in case of address change or user session initialization.
    *
+   * @param options
+   * @param options.selectedAddress - the selectedAddress against which to detect for token balances
+   * @param options.chainId - the chainId against which to detect for token balances
    */
-  restartTokenDetection() {
-    if (!(this.isActive && this.selectedAddress)) {
+  restartTokenDetection({ selectedAddress, chainId } = {}) {
+    const addressAgainstWhichToDetect = selectedAddress ?? this.selectedAddress;
+    const chainIdAgainstWhichToDetect = chainId ?? this.chainId;
+    if (
+      !(
+        this.isActive ||
+        !addressAgainstWhichToDetect ||
+        chainIdAgainstWhichToDetect !== CHAIN_IDS.MAINNET
```

Comment:
Currently, token detection is enabled for only MAINNET but on Binance, Polygon, and Avalanche. We are checking for the same in line 97 as well.

Reply:
May be use=ing method `isTokenDetectionEnabledForNetwork` will be better here than this comparison.

Reply:
true I suppose we can just remove this check since it takes place in `detectNewTokens` which is called below.

Reply:
updated [here](https://github.com/MetaMask/metamask-extension/pull/16866/commits/34af9a7e31b45602662c3b1f802f3022cc1f99b2) @NiranjanaBinoy